orchestrator/domain/base.py

```diff
@@ -481,6 +485,7 @@ def dict(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
         return attribs
 
 
+# class ProductBlockModelMeta(type(BaseModel)):  # type: ignore[misc]
 class ProductBlockModelMeta(ModelMetaclass):
```

Comment:
I thought that with `class ProductBlockModelMeta(type(BaseModel))` I had a workaround for importing Pydantic's internal `ModelMetaClass`. 

Technically it works but mypy doesn't recognize it, hence the `type: ignore[misc]`. And as it turns out we'll also have to set the ignore on each and every class inheriting from `ProductBlockModel` - that, or add `metaclass=ProductBlockModelMeta` to the class constructor.

I don't think there's a way that we can extend Pydantic's existing metaclass without some downsides.

But, while importing Pydantic's internal metaclass is dirty, we're not relying on anything it does - it's just being formed as a _hook_ of sorts. So it's only slightly terrible. :)

Reply:
Solved by removing ProductBlockModelMeta and moving _fix_pb_data into ProductBlockModel.